modules/schema/src/main/scala/lucuma/odb/json/time.scala

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.json
+
+import cats.syntax.bifunctor.*
+import cats.syntax.either.*
+import io.circe.Codec
+import io.circe.Decoder
+import io.circe.DecodingFailure
+import io.circe.Encoder
+import io.circe.HCursor
+import io.circe.Json
+import io.circe.refined._
+import io.circe.syntax._
+
+import java.math.RoundingMode.DOWN
+import java.time.Duration
+import java.time.temporal.ChronoUnit.MICROS
+import scala.util.Try
+
+object time {
+
+  // N.B. We cannot represent any random `Duration` with the existing GraphQL
+  // schema.  Only a Duration expressed as a `Long` in milliseconds can be
+  // represented.  We need to either change the schema or make a specialized
+  // duration type.
```

Comment:
Regardless of the fate of this PR, I'd like to revisit our use of `Duration` in another pass.  We're storing exposure times as an `interval` which doesn't align with `java.time.Duration` (and which need a non-negative constraint as well).  We also encode `Duration` as a time in microseconds as a `Long` value, but that also doesn't align.  This caused me pain and seems like it will continue to produce pain.  I'm going to try an `Interval` type along the lines of our `Timestamp`, I think.